Makefile.am

```diff
@@ -30,112 +31,124 @@ bin_PROGRAMS = \
 	ported/librdmacm/fi_cmatose \
 	complex/fabtest
 
+common_sources = common/shared.c
+
+if MACOS
+common_sources += common/osx/osd.h
+common_sources += common/osx/osd.c
+endif
+
 simple_fi_info_SOURCES = \
 	simple/info.c \
-	common/shared.c
+	$(common_sources)
```

Comment:
Since there are now potentially 2 .c files that are added to every test program, it probably now makes sense to have a libfabtests.la convenience library that is linked in to every executable (rather than potentially re-compiling the same sources over and over and over and ...).

You can have `libfabtests_la_SOURCES = common/shared.c` and still `+=` to it in the "if MACOS" case, just like you have now.  Then add a LIBADD (or LIBS? I never remember which is the right one offhand...) to get libfabtests.la linked to each test program.